### DISCUSSION TITLE
Spoke making mutiple copies of the same scene with different SceneIDs. How does it work?

### DISCUSSION THREAD

**Discussion by Charlesc22** [mozilla/hubs#2932] [General Questions & Discussion] [answered]
Can someone clarify how updating a scene works? I have an existing room using scene A. I update scene A and keep the "Publish name" the same in spoke and publish it. but when I re-load the room is still uses the previous version of the scene
This has baffled me as some times it updates the scene and sometimes it doesn't
I went in my admin panel and it shows 2 different scenes with the same name
and 2 different sceneIDs
I actually have 3 versions, all with the same name, but 3 different sceneIDs. When does spoke decide to generate a new sceneID vs re-use the same one?

**Comment by emclaren** [comment, marked as answer]
Hi @Charlesc22 

A room that already exists and has a scene should automatically updated you republish that scene.* For example if you have a room with an empty crater scene, if you edit your crater scene to include a house object, when you re- open the room, you will see the original empty crater. 

To update to the most recent version without leaving the room. You need to go to the dropdown menu, select "choose scene" and select the updated version in your lists of scenes (or you can type /scene scene-url-placeholder into the text chat).

A scene will have the same URL persistently. Every time you work on a scene and republish it, you can open up the scene landing page with a URL like this https://hubs.mozilla.com**/scenes/**CYep2GP  (note the word "scenes") - scene landing pages look like this: 
<img width="1663" alt="Screen Shot 2020-09-10 at 2 48 19 PM" src="https://user-images.githubusercontent.com/6835324/92813323-d2b4aa00-f375-11ea-8b83-a3aef1134dc1.png">

If you click on "Create Room with This Scene" - it will create a brand new room (with a unique URL) every time you press this button. If you want to keep the same room URL and update scene, just visit your existing room and change the scene from the dropdown (as described above).

To my knowledge the only occasion when a scene URL should change is if choose to make a copy. If you go to the dropdown menu in spoke and click "save as" - this will create a duplicate of your scene with a new URL. This can be helpful for version control if you want to play around with a scene without worrying about losing the original design. 

* Note I updated this because it turns out I was mistaken - a room will automatically update when the scene is updated

**Reply by navratiljudit** [reply]
Thank you for this question and response. 
Is there a way to keep editing the scene in Spoke but use the same URL? and not having to change the scene but to update the scene in spoke?

THank you!

**Reply by emclaren** [reply]
as long as you don't click "save as" your scene will keep publishing to the same URL as you update it. 

If you want to work on the scene in spoke without the published scene changing you can select "save" from the dropdown menu. 

I hope this answers your question. Let me know if I didnt understand correctly.

**Reply by robinkwilson** [reply]
Marking as answer to get this out of our pipeline! @ me if this needs to be reopened.
